rclpy/test/test_node.py

```diff
@@ -838,6 +838,134 @@ def test_node_set_parameters_rejection(self):
         self.assertIsInstance(result[0], SetParametersResult)
         self.assertFalse(result[0].successful)
 
+    def reject_parameter_callback_1(self, parameter_list):
+        rejected_parameters = (
+            param for param in parameter_list if 'reject1' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def reject_parameter_callback_2(self, parameter_list):
+        rejected_parameters = (
+            param for param in parameter_list if 'reject2' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def reject_parameter_callback_3(self, parameter_list):
+        rejected_parameters = (
+            param for param in parameter_list if 'reject3' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def test_node_set_paramters_rejection_list(self):
+        # Declare a new paramters and set a list of callbacks so that it's rejected when set.
+        reject_list_parameter_tuple = [
+            (
+                'reject1', True, ParameterDescriptor()
+            ),
+            (
+                'non_reject', True, ParameterDescriptor()
+            ),
+            (
+                'non_reject', True, ParameterDescriptor()
+            )
+        ]
+
+        self.node.declare_parameters('', reject_list_parameter_tuple)
+        self.node.add_on_set_parameters_callback(
+            self.reject_parameter_callback_1)
```

Comment:
Similar to what @dirk-thomas wrote above, I see no reason for this not to be on the same line as the add call.

Reply:
@allenh1 sir could you explain a liitle bit..in this?

Reply:
I'm saying that this:

```py
 self.node.add_on_set_parameters_callback(
            self.reject_parameter_callback_1)
```

Has no reason to not just be this:

```
 self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
```

Reply:
@allenh1 done sir